### ISSUE TITLE
`jnp.nanpercentile` does not work correctly after `jnp.nanmean`

### ISSUE THREAD

**Issue by sergiuiacob1** [google/jax#7894]
## Replicating the issue

```python
import jax
import jax.numpy as jnp

from jax.config import config
config.update("jax_enable_x64", True)

percentile = lambda data: jnp.nanpercentile(data, 10, axis=1)
percentile = jax.jit(percentile)
mean = lambda data: jnp.nanmean(data, axis=0, keepdims=True)
mean = jax.jit(mean)

# This works
data = jnp.array([[5.5, 6.25, 5.5, 7.25, jnp.nan]], dtype=jnp.float64)
data = data.reshape((1, 5))

print(data.shape, data)
data = percentile(data)
print(data.shape, data)


# This does not work (returns nan)
print("\n\n\n")
data = jnp.array([[5.5, 6.25, 5.5, 7.25, jnp.nan], [5.5, 6.25, 5.5, 7.25, jnp.nan]], dtype=jnp.float64)
data = data.reshape((2, 5))

print(data.shape, data)
data = mean(data)
print(data.shape, data)
data = percentile(data)
print(data.shape, data)
```

## Code output
```
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
(1, 5) [[5.5  6.25 5.5  7.25  nan]]
(1,) [5.5]




(2, 5) [[5.5  6.25 5.5  7.25  nan]
 [5.5  6.25 5.5  7.25  nan]]
(1, 5) [[5.5  6.25 5.5  7.25  nan]]
(1,) [nan]
```

## Expected result
I'm expecting the second piece of code to return the same value – `5.5`. The second part contains a `nanmean` step, which should compute average values per columns. Since the values are the same on both rows, the `nanmean` result should be `[5.5, 6.25, 5.5, 7.25, nan]`.

If I apply my `percentile` function on the data directly, it works. If I apply it after running the `mean` function on the data, then it doesn't work.

**Comment by jakevdp** [comment]
Thanks for the report – what version of JAX are you using? I don't see this issue in JAX v0.2.19, and this looks similar to an issue that was fixed in #6439 (which made it into version 0.2.13)

**Comment by sergiuiacob1** [comment]
> Thanks for the report – what version of JAX are you using? I don't see this issue in JAX v0.2.19, and this looks similar to an issue that was fixed in #6439 (which made it into version 0.2.13)

I'm using `0.2.9`. I'm indeed not seeing this issue with version `0.2.13`. I'll close this since it has been fixed in a later version.
